src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -1067,18 +1067,21 @@ export const GlobalProvider = memo(
                         deriveId,
                         deriveId
                     );
-                    const derivedId = deriveId(id);
+                    const derivedIds = nodeIds.map((id) => deriveId(id));
                     newNodes.forEach((n) => {
                         // eslint-disable-next-line no-param-reassign
-                        n.selected = n.id === derivedId;
+                        n.selected = derivedIds.includes(n.id);
                     });
                     return [...setSelected(nodes, false), ...newNodes];
                 });
 
                 changeEdges((edges) => {
                     const newEdge = copyEdges(
                         edges.filter((e) => {
-                            return nodesToCopy.has(e.target) && nodesToCopy.has(e.source);
+                            return (
+                                (nodesToCopy.has(e.target) && nodesToCopy.has(e.source)) ||
+                                edgeIds.includes(e.id)
```

Comment:
Question: Users are able to select arbitrary edges. What will this do when the user selects a single edge and nothing else? What will the result of the user copying that be?

More general: How is copying edges supposed to behave?

Reply:
When i tested it, nothing happened (as expected) but i suppose it's possible it was still duplicating it. I didn't see any errors in the console though

Reply:
Try this:
1. Create a chain with a lest one edge.
2. Select one edge and nothing else.
3. <kbd>Ctrl+D</kbd>
4. You will see that the edge you previously selected was deselected. Select it again by clicking on it.
5. Press <kbd>DEL</kbd>
6. You will see that the edge will be deselected.

What actually happened is that we duplicated the edge and then removed the duplicate again.

Reply:
Ah. I'll fix that then